Core/Src/main.c

```diff
@@ -95,19 +96,61 @@ int main(void)
   MX_ADC_Init();
   MX_TIM16_Init();
   MX_USART1_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
 
   /* USER CODE BEGIN 2 */
-  debug("\n\nBootcamp starting up...");
-  debug("Compiled on %s at %s", __DATE__, __TIME__);
 
+  //start pwm
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
+
+  //define buffers and variables
+  uint8_t spi_recieve_buf[10];
+  uint8_t spi_transmit_buf[10];
+  uint16_t TRANSMIT_RECIEVE_SIZE = 3;
```

Comment:
This TRANSMIT_RECIEVE_SIZE variable can be removed using the sizeof() function on the buffer. That also makes it one fewer change if ever the length of the message being sent changes (which it won't for this).